www/perl-lib/SMT/Registration.pm

```diff
@@ -51,6 +53,28 @@ sub handler {
         $r->log_error("protocol version '".$hargs->{'version'}."' not implemented");
         return SMT::Utils::http_fail($r, 400, "Invalid protocol version.");
     }
+
+    # check if registration sharing is enabled and load module if it is
+    if (! $REGSHARING) {
+        $REGSHARING = SMT::Utils::hasRegSharing($r);
```

Comment:
If hasRegSharing returns 0/1 you can call the function directly everywhere where you need it.
If this function caches the value, you do not need to cache it again.